custom_components/fraimic/library.py

```
@@ -207,6 +207,29 @@ async def async_update_image(
+        """Rename an original while preserving its stable library id."""
+        image = self.get(image_id)
+        filename = safe_filename(filename)
+        if filename == image.filename:
+            return image
+
+        old_path = self.original_path(image)
+        new_path = self.originals_dir / f"{image.image_id}_{filename}"
+        try:
+            await self.hass.async_add_executor_job(old_path.replace, new_path)
+        except OSError as err:
+            raise HomeAssistantError(f"Could not rename image: {err}") from err
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)